packages/mobile-app/data/api/walletServer.ts

```diff
@@ -16,8 +16,23 @@ const WALLET_SERVER_URLS: Record<Network, string> = {
 class WalletServer {
   transformers: WalletServerTransformer[] = [ForkTester];
 
+  private latestBlockCache: Map<
+    Network,
+    { time: number; response: GetLatestBlockResponse }
+  > = new Map();
+  private LATEST_BLOCK_CACHE_MS = 5000;
+
   async getLatestBlock(network: Network): Promise<GetLatestBlockResponse> {
     const url = WALLET_SERVER_URLS[network] + "latest-block";
+
+    const cached = this.latestBlockCache.get(network);
+    if (
+      cached &&
+      performance.now() - cached.time < this.LATEST_BLOCK_CACHE_MS
+    ) {
+      return cached.response;
+    }
+
```

Comment:
Not sure if you are interested, but you can generate typesafe api client from the openapi specification, could be nice because it will force the API to stay in sync

Reply:
I'm open to it, but it's not high on the priority list until the app functionality is built out. When I looked at it earlier, I was running into some compatibility issues with the generator output and React Native, so I skipped it for the time being